docs/docs/guides/labs/dg/python-environment-management-and-uv-integration.md

```diff
@@ -16,21 +16,45 @@ is likely to change in the next few releases.
-
-If the setting is `active`, then `dg` does nothing fancy. It is up to the user to manage their own Python environments. If individual project-scoped environments are desired, the user must create them, and ensure the appropriate one is activated when running `dg` commands against that project.
-
-If the setting is `persistent_uv`, then `dg` will ignore any activated virtual environments. Subprocesses will be launched using `uv run`, which handles resolution of the project-scoped `.venv` environment and ensures subprocesses are launched in that environment. It also ensures that the environment is up to date with the project's specified dependencies before launching a subprocess.
-
-The `persistent_uv` setting is intended to ease development friction by removing the burden of juggling virtual environments from the user. It is our recommended configuration. However, we are still developing its functionality, and some users may prefer to totally opt out of any environment management "magic". These users should make sure all their projects use `tool.dg.project.python_environment = "active"`.
-
-:::info
-
-Disabling `uv` integration by setting `tool.dg.project.python_environment
-= "active"` will currently disable the `dg cache`. This will make some
-operations noticeably slower.
-
-:::
+The Python environment used for a project is determined by the `tool.dg.project.python_environment` setting in the project `pyproject.toml` file. This setting is either `active` (the default) or `uv_managed`:
+
+<Tabs>
+    <TabItem value="active" label="active">
+        <Tabs>
+        <TabItem value="dg.toml" label="dg.toml">
+                ```
+                [project.python_environment]
+                active = true 
+                ```
+        </TabItem>
+        <TabItem value="pyproject.toml" label="pyproject.toml">
+                ```
+                [tool.dg.project.python_environment]
+                active = true 
+                ```
+        </TabItem>
+        </Tabs>
+    </TabItem>
+    <TabItem value="uv_managed" label="uv_managed">
+        <Tabs>
+            <TabItem value="dg.toml" label="dg.toml">
+                ```
+                [project.python_environment]
+                uv_managed = true 
+                ```
+            </TabItem>
+            <TabItem value="pyproject.toml" label="pyproject.toml">
+                ```
+                [tool.dg.project.python_environment]
+                uv_managed = true 
+                ```
+            </TabItem>
+        </Tabs>
+    </TabItem>
+</Tabs>
+
+If the setting is `active` (the default), then it is up to the user to manage their own Python environments. If individual project-scoped environments are desired, the user must create them, and ensure the appropriate one is activated when running `dg` commands against that project.
```

Comment:
i think its worth talking through how we prompt to `uv sync` in this configuration and what the thinking behind that is